### PR TITLE
Close ESC Menu on Death (#371)

### LINKED ISSUE #371: Escape Menu should close on "death"
<Please review the guidelines for contributing to this repository. The link is above.>
<If you are reporting a bug then you must use the provided issue template.>

### Expected behaviour
Escape menu should close when dying to prevent preemptive dead logging (there's a block to stop opening the escape menu whilst dead so why not close it when they die)

### Actual behaviour
escape menu stays up allowing player to logout whilst bleeding out

### Steps to reproduce the behaviour 
jumped of building, hit escape, land on floor, break legs scream in agony,wait for 10 second timer to expire (for combat logging), click exit game or w/e

----

Mission version: 5.0 and probably prior

Branch: dev

### PR DESCRIPTION
**Pull request by BlackFisch**
Resolves #371

#### Changes proposed in this pull request: 
- fn_onPlayerKilled.sqf: Close ESC Menu

### CODE REVIEW

**Comment by BlackFisch** [comment]
fix not working, my bad

**Comment by Jawshy** [comment]
Try that code in ../Altis_Life.Altis/core/functions/[fn_escInterupt.sqf](https://github.com/AsYetUntitled/Framework/blob/7784f81aac515a5670cb7379cf367c117665e00c/Altis_Life.Altis/core/functions/fn_escInterupt.sqf) in the [for](https://community.bistudio.com/wiki?title=for&oldid=99336) operator, @BlackFisch and @RGoldhardt.

**Comment by BlackFisch** [comment]
You mean asking for the player to be dead? actually a good idea... i'll try that later! Tanks :)

**Comment by BlackFisch** [comment]
ToDo: Change "for" in fn_escInterup.sqf

**Comment by Jawshy** [comment]
@imKLM and I were discussing this in `#development` on the _ArmA RPG Life_ Discord server. You're welcome to [join](https://discord.gg/EaB7Jgw).

**Comment by BlackFisch** [comment]
Thanks for the invite @Jawshy - I honestly don't use discord very much, that's why I didn't thought of that ^^

**Comment by BlackFisch** [comment]
Okay, added and tested the code - should work just fine now :) I also edited the code in the fn_onPlayerKilled.sqf to close all other existing dialogs (like virtual inventory and stuff)

File: Altis_Life.Altis/core/functions/fn_escInterupt.sqf [hunk 1]
```diff
@@ -74,5 +74,8 @@ for "_i" from 0 to 1 step 0 do {
     if (_usebleCtrl) then {
         _respawnButton ctrlEnable true; //Enable the button.
     };
-    waitUntil {isNull (findDisplay 49)};
+    waitUntil {isNull (findDisplay 49) || {!alive player}};
+    if (!isNull (findDisplay 49)) then {
```
Comment (BoGuu) [inline Altis_Life.Altis/core/functions/fn_escInterupt.sqf]: Although unlikely, they could *possibly* reopen the menu after closing it (while alive), causing it to be closed again. Worth adding another dead check in the !isNull if statement. Will commit, for you.

**Comment by BoGuu** [comment]
Thanks! :)

**Pull request merged**